crates/extensions/c8y_auth_proxy/src/server.rs

```diff
@@ -225,6 +226,7 @@ async fn connect_to_websocket(
     req = req.header("Authorization", format!("Bearer {token}"));
     let req = req
         .uri(uri)
+        .header(HOST, url::Url::parse(uri).expect("URI must have valid host to reach c8y proxy").host_str().unwrap())
```

Comment:
There is likely a tidier way to achieve this, I'll work on it while I try and add some unit tests. From what I can tell, `tokio_tungstenite` requires us to add the host header in manually, which was why the unit tests were failing after removing the incorrect host header copying as we then had no host header for outgoing websocket requests.